api/src/services/opportunities_v1/get_opportunity.py

```diff
@@ -27,11 +25,21 @@ def _fetch_opportunity(
         # we need to explicitly join here as the "*" approach doesn't
         # seem to work with the way our agency relationships are setup
         .options(selectinload(Opportunity.agency_record).selectinload(Agency.top_level_agency))
+        # Do not load the following relationships, they aren't necessary for
+        # our opportunity endpoints, and would make the query much larger/slower
+        # if we were to fetch them.
+        # This effectively undoes the `selectinload("*")` above for these relationships
+        # and makes them lazily loaded (the default for relationships) - keeping them out of the query entirely.
+        .options(
+            lazyload(Opportunity.opportunity_change_audit),
+            lazyload(Opportunity.all_opportunity_summaries),
+            lazyload(Opportunity.all_opportunity_notification_logs),
+            lazyload(Opportunity.saved_opportunities_by_users),
+            lazyload(Opportunity.competitions),
+            lazyload(Opportunity.versions),
+        )
     )
 
-    if not load_all_opportunity_summaries:
-        stmt = stmt.options(noload(Opportunity.all_opportunity_summaries))
```

Comment:
FYI - I got rid of `noload` because the docs recommend never using it - it exists for legacy reasons. It makes it so the value is forced to be `None` and has weird behaviors. It's better to just change set things to `lazyload` - worst case there is we try to fetch something that has to be lazyloaded.